src/Meta/MetaFactory.cpp

```diff
@@ -193,15 +193,14 @@ void MetaFactory::createFromFunction(const clang::FunctionDecl& function, Functi
     bool returnsNotRetained = Utils::getAttributes<clang::NSReturnsNotRetainedAttr>(function).size() > 0 || Utils::getAttributes<clang::CFReturnsNotRetainedAttr>(function).size() > 0;
 
     // Clang doesn't handle The Create Rule automatically like for methods, so we have to do it manually
-    if (!(returnsRetained || returnsNotRetained)) {
-        Type* returnType = functionMeta.signature[0];
-        if (returnType->is(TypeInterface) || returnType->is(TypeBridgedInterface) || returnType->is(TypeId)) {
-            std::string functionName = function.getNameAsString();
+    if (!(returnsRetained || returnsNotRetained) && functionMeta.signature[0]->is(TypeBridgedInterface)) {
```

Comment:
We didn't find a possible scenario where we should explicitly memory manage something that is ```TypeInterface``` or ```TypeId```. 
I experimented and created a method that returns a ```dispatch_queue_t``` and it seems that it is returned autoreleased so should we memory manage it ?